lms/static/js/components/card/views/card.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * A generic card view class.
+ *
+ * Subclasses can override any of the following:
+ * - configuration (string or function): Sets the display style of the card as square or list. Can take values of
+ *      "square_card" or "list_card". Defaults to "square_card".
+ * - action (function): Action to take when the action button is clicked. Defaults to a no-op.
+ * - cardClass (string or function): Class name for this card's DOM element. Defaults to the empty string.
+ * - title (string or function): Title of the card. Defaults to the empty string.
+ * - description (string or function): Description of the card. Defaults to the empty string.
+ * - details (array or function): Array of child views to be rendered as details of this card. The class "meta-detail"
+ *      is automatically added to each rendered child view to ensure appropriate styling. Defaults to an empty list.
+ * - actionClass (string or function): Class name for the action DOM element. Defaults to the empty string.
+ * - actionUrl (string or function): URL to navigate to when the action button is clicked. Defaults to the empty string.
+ * - actionContent: Content of the action button. This may include HTML. Default to the empty string.
+ */
+;(function (define) {
+    'use strict';
+    define(['jquery',
+            'backbone',
+            'text!templates/components/card/square_card.underscore',
+            'text!templates/components/card/list_card.underscore'],
+        function ($, Backbone, squareCardTemplate, listCardTemplate) {
+            var CardView = Backbone.View.extend({
+                events: {
+                    'click .action' : 'action'
+                },
+
+                switchOnConfiguration: function (square_result, list_result) {
+                    return this.callIfFunction(this.configuration) === 'square_card' ?
+                        square_result : list_result;
+                },
+
+                callIfFunction: function (value) {
+                    if ($.isFunction(value)) {
+                        return value.call(this);
+                    } else {
+                        return value;
+                    }
+                },
+
+                constructor: function (options) {
```

Comment:
initialize seems to run last in the initialization process, after functions like className. className depends on configuration being set in order to choose the appropriate class, so configuration must be set in the constructor instead.